src/server/game/Groups/Group.cpp

```diff
@@ -1875,6 +1875,10 @@ GroupJoinBattlegroundResult Group::CanJoinBattlegroundQueue(Battleground const*
         // don't let join to bg queue random if someone from the group is already in bg queue
         if (bgTemplate->GetBgTypeID() == BATTLEGROUND_RB && member->InBattlegroundQueue())
             return ERR_IN_NON_RANDOM_BG;
+
+        // don't let Death Knights join BG queues when they are not allowed to be teleported yet
+        if (member->getClass() == CLASS_DEATH_KNIGHT && member->GetMapId() == 609 && !member->IsGameMaster() && !member->HasSpell(50977))
+            return ERR_BATTLEGROUND_JOIN_TIMED_OUT;
```

Comment:
For groups, this error should be used. This include not eligible message if a DK is in the party.
```suggestion
            return ERR_GROUP_JOIN_BATTLEGROUND_FAIL;
```